ERCS/erc-7857.md

```diff
@@ -0,0 +1,318 @@
+/// @return _tokenId The ID of the newly minted token
+function mint(bytes[] calldata _proofs, string[] calldata _dataDescriptions)
+    external
+    payable
+    returns (uint256 _tokenId);
+
+/// @notice Transfer full data (full means data and ownership)
+/// @param _to Address to transfer data to
+/// @param _tokenId The token to transfer data for
+/// @param _proofs Proofs of data available for _to
+function transfer(
+    address _to,
+    uint256 _tokenId,
+    bytes[] calldata _proofs
+) external;
+
+/// @notice Clone data (clone means just data, not ownership)
+/// @param _to Address to clone data to
+/// @param _tokenId The token to clone data for
+/// @param _proofs Proofs of data available for _to
+/// @return _newTokenId The ID of the newly cloned token
+function clone(
+    address _to,
+    uint256 _tokenId,
+    bytes[] calldata _proofs
+) external payable returns (uint256 _newTokenId);
+
+/// @notice Transfer public data with ownership
+/// @param _to Address to transfer data to
+/// @param _tokenId The token to transfer data for
+function transferPublic(
+    address _to,
+    uint256 _tokenId
+) external;
+
+/// @notice Clone public data
+/// @param _to Address to clone data to
+/// @param _tokenId The token to clone data for
+/// @return _newTokenId The ID of the newly cloned token
+function clonePublic(
+    address _to,
+    uint256 _tokenId
+) external payable returns (uint256 _newTokenId);
+
+/// @notice Add authorized user to group
+/// @param _tokenId The token to add to group
+function authorizeUsage(
+    uint256 _tokenId,
+    address _user
+) external;
+
+/// @notice Get token owner
+/// @param _tokenId The token identifier
+/// @return The current owner of the token
+function ownerOf(uint256 _tokenId) external view returns (address);
+
+/// @notice Get the authorized users of a token
+/// @param _tokenId The token identifier
+/// @return The current authorized users of the token
+function authorizedUsersOf(uint256 _tokenId) external view returns (address[] memory);
+```
+
+## Rationale
+
+The design choices in this standard are motivated by several key requirements:
+
+1. **Verification Abstraction**: The standard separates the verification logic into a dedicated interface (IDataVerifier), allowing different verification mechanisms (TEE, ZKP) to be implemented and used interchangeably. The verifier should support two types of proof:
+    - Ownership Proof Verifies that the prover possesses the original data by demonstrating knowledge of the pre-images that generate the claimed dataHashes
+    - Transfer Validity Proof Verifies secure data integrity and availability by proving: knowledge of the original data (pre-images of oldDataHashes); ability to decrypt with oldKey and re-encrypt with newKey; secure transmission of newKey using recipient's public key; integrity of the newly encrypted data matching newDataHashes; and data availability confirmed by recipient's signature on both oldDataHashes and newDataHashes
```

Comment:
```suggestion
    - Transfer Validity Proof Verifies secure data integrity and availability by proving: knowledge of the original data (pre-images of oldDataHashes); ability to decrypt with oldKey and re-encrypt with newKey; secure transmission of newKey using recipient's public key; integrity of the newly encrypted data matching newDataHashes; and data availability confirmed by recipient's signature on both `oldDataHashes` and `newDataHashes`
```

I am no doubt going to miss some of these inline code snippets. They all should be in backticks.